src/keria/app/aiding.py

```diff
@@ -245,6 +248,110 @@ def anchorSeals(agent, ixn):
+        
+        body = req.get_media()
+        
+        if body.get("approveDelegation") is not None:
+            op = self.approveDelegation(agent, name, body)
+        else:
+            raise falcon.HTTPBadRequest(title="invalid request",
+                                        description=f"required field 'approveDelegation' missing from request")
+
+        try:
+            rep.status = falcon.HTTP_200
+            rep.content_type = "application/json"
+            rep.data = op.to_json().encode("utf-8")
+        except (kering.AuthError, ValueError) as e:
+            raise falcon.HTTPBadRequest(description=e.args[0])
+        except (falcon.HTTPBadRequest, falcon.HTTPNotFound, falcon.HTTPInternalServerError) as e:
+            raise e
+
+    # def on_delete(self, req, rep, name):
+    #     """ Identifier approval delete endpoint not currently implemented
+
+    #     Parameters:
+    #         req (Request): falcon.Request HTTP request object
+    #         rep (Response): falcon.Response HTTP response object
+    #         name (str): human readable name for Hab to delete
+
+    #     """
+    #     raise falcon.HTTPBadRequest(description="Identifier approval endpoint DELETE not implemented")
+
+    # def on_post(self, req, rep, name):
+    #     """ Identifier approval POST functionality not currently implemented
+
+    #     Parameters:
+    #         req (Request): falcon.Request HTTP request object
+    #         rep (Response): falcon.Response HTTP response object
+    #         name (str): human readable name for Hab to rotate or interact
+
+    #     """
+    #     raise falcon.HTTPBadRequest(description="Identifier approval endpoint POST not implemented")
+    
+    @staticmethod
+    def approveDelegation(agent, name, body):
+        hab = agent.hby.habByName(name)
+        if hab is None:
+            raise falcon.HTTPNotFound(title=f"No AID {name} found")
+
+        ixn = body.get("approveDelegation")
+        if ixn is None:
+            raise falcon.HTTPBadRequest(title="invalid interaction",
+                                        description=f"required field 'approveDelegation' missing from request")
+
+        sigs = body.get("sigs")
+        if sigs is None or len(sigs) == 0:
+            raise falcon.HTTPBadRequest(title="invalid interaction",
+                                        description=f"required field 'sigs' missing from approveDelegation request")
+
+        serder = serdering.SerderKERI(sad=ixn)
+        
+        gatePre = ixn['a'][0]['i']
+        gateSaid = ixn['a'][0]['d']
+
+        for (pre, sn), dig in hab.db.delegables.getItemIter():
+            if pre == gatePre:
+                seqner = coring.Seqner(sn=serder.sn)
+                couple = seqner.qb64b + serder.saidb
+                dgkey = dbing.dgKey(coring.Saider(qb64=gatePre).qb64b, coring.Saider(qb64=gateSaid).qb64b)
+                hab.db.setAes(dgkey, couple)  # authorizer event seal (delegator/issuer)
+
```

Comment:
You need to parse the DIP event to get it into your kevers.  Just creating the Aes isn't sufficient.

Reply:
The dip event is waiting in escrows to be processed but the one dependency needed is Aes. Based on our previous discussions I assumed this was the proper solution to allow the processing. I'll adjust it however you prefer.